orc/src/main/java/org/apache/iceberg/orc/OrcRowReader.java

```diff
@@ -29,6 +29,6 @@
   /**
    * Reads a row.
    */
-  T read(VectorizedRowBatch batch, int row);
+  T read(VectorizedRowBatch batch, long batchOffsetInFile, int rowOffsetInBatch);
```

Comment:
This seems to introduce a lot of code churn, when most implementations don't use `batchOffsetInFile`. What about a less intrusive way of passing this by using a context method that is called once for each batch?

Parquet has something similar, where each row group causes new context to be passed to the readers: https://github.com/apache/iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReader.java#L32

This could expose a method like `setBatchContext(long batchOffsetInFile)` with a no-op default. Then only a few implementations would need to change.

Reply:
Thanks for the suggestion! This reduces the code changes significantly.